sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -116,30 +116,38 @@ def _hdbscan_brute(
     **metric_params,
 ):
     if metric == "precomputed":
-        # Treating this case explicitly, instead of letting
-        # sklearn.metrics.pairwise_distances handle it,
-        # enables the usage of numpy.inf in the distance
-        # matrix to indicate missing distance information.
+        if X.shape[0] != X.shape[1]:
+            raise ValueError(
+                "The precomputed distance matrix is expected to be symmetric, however"
+                f" it has shape {X.shape}. Please verify that the"
+                " distance matrix was constructed correctly."
+            )
+        if not _allclose_dense_sparse(X, X.T):
+            raise ValueError(
+                "The precomputed distance matrix is expected to be symmetric, however"
+                " its values appear to be asymmetric. Please verify that the distance"
+                " matrix was constructed correctly."
+            )
+
         distance_matrix = X.copy() if copy else X
     else:
         distance_matrix = pairwise_distances(
             X, metric=metric, n_jobs=n_jobs, **metric_params
         )
     distance_matrix /= alpha
 
-    # max_dist is only relevant for sparse and is ignored for dense
-    max_dist = metric_params.get("max_dist", 0.0)
-    sparse = issparse(distance_matrix)
-    distance_matrix = distance_matrix.tolil() if sparse else distance_matrix
+    max_distance = metric_params.get("max_distance", 0.0)
+    if issparse(distance_matrix) and distance_matrix.format != "csr":
```

Comment:
[`scipy.sparse.isspmatrix_csr`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.isspmatrix_csr.html#scipy.sparse.isspmatrix_csr) exists ; and is probably preferable if someone extends and uses `scipy.sparse.csr_matrix`?

```suggestion
    if isspmatrix_csr(distance_matrix):
```